src/AppInstallerCLICore/ExecutionContext.h

```diff
@@ -194,6 +202,12 @@ namespace AppInstaller::CLI::Execution
         {
             m_data[D].emplace<details::DataIndex(D)>(std::forward<typename details::DataMapping<D>::value_t>(v));
         }
+        template <Data D>
+        void Add(const typename details::DataMapping<D>::value_t& v)
+        {
+            auto copy = v;
+            m_data[D].emplace<details::DataIndex(D)>(std::forward<typename details::DataMapping<D>::value_t>(std::move(copy)));

```

Comment:
>std::move(copy) [](start = 109, length = 15)

You shouldn't need to move a copy; why can't you just emplace v and have it do the copy?